build.gradle

```diff
@@ -1,153 +1,84 @@
-  modImplementation include(fabricApi.module("fabric-renderer-registries-v1", fabricVersion))
-  modImplementation include(fabricApi.module("fabric-rendering-v1", fabricVersion))
-  modImplementation include(fabricApi.module("fabric-resource-loader-v0", fabricVersion))
-  modImplementation include(fabricApi.module("fabric-screen-handler-api-v1", fabricVersion))
+	// Fabric API. This is technically optional, but you probably want it anyway.
+	modImplementation "net.fabricmc.fabric-api:fabric-api:${project.fabric_version}"
 
-  modImplementation "com.terraformersmc:modmenu:1.16.23"
+	// Uncomment the following line to enable the deprecated Fabric API modules. 
+	// These are included in the Fabric API production distribution and allow you to update your mod to the latest modules at a later more convenient time.
 
-  implementation "com.google.code.findbugs:jsr305:3.0.2"
-  implementation "org.jetbrains:annotations:23.0.0"
-  implementation 'org.checkerframework:checker-qual:3.22.1'
+	// modImplementation "net.fabricmc.fabric-api:fabric-api-deprecated:${project.fabric_version}"
 }
 
-compileJava {
-  options.with {
-    it.deprecation = true
-    it.encoding = "UTF-8"
-    it.fork = true
-    it.compilerArgs.addAll(
-        ["-Xlint:all", "-Xlint:-processing",
-         // Enable parameter name class metadata 
-         // https://openjdk.java.net/jeps/118
-         "-parameters"]
-    )
-    it.release.set(8)
-  }
+base {
+	archivesName = project.archives_base_name
 }
 
 processResources {
-  filesMatching("/fabric.mod.json") {
-    expand version: version
-  }
-}
-
-jar {
-  from "/LICENSE"
+	inputs.property "version", project.version
 
-  manifest.attributes(
-      "Build-Timestamp": Instant.now(),
-      "Build-Revision": versioning.info.commit,
-      "Build-Jvm": ""
-          + "${System.getProperty("java.version")} ("
-          + "${System.getProperty("java.vm.vendor")} "
-          + "${System.getProperty("java.vm.version")})",
-      "Built-By": GradleVersion.current(),
-
-      "Implementation-Title": project.name,
-      "Implementation-Version": project.version,
-      "Implementation-Vendor": project.group,
-
-      "Specification-Title": "FabricMod",
-      "Specification-Version": "1.0.0",
-      "Specification-Vendor": "dev.sapphic",
-
-      "Sealed": "true"
-  )
+	filesMatching("fabric.mod.json") {
+		expand "version": project.version
+	}
 }
 
-def antSignJar(task) {
-  for (final file in task.outputs.files) {
-    ant.signjar(
-        jar: file,
-        alias: property("signing.mods.keyalias"),
-        storepass: property("signing.mods.password"),
-        keystore: property("signing.mods.keystore"),
-        verbose: true,
-        preservelastmodified: true
-    )
-  }
+tasks.withType(JavaCompile).configureEach {
+	// Minecraft 1.18 (1.18-pre2) upwards uses Java 17.
+	it.options.release = 17
 }
 
-if (hasProperty "signing.mods.keyalias") {
```

Comment:
Why was the optional JAR signing removed?

Reply:
Gradle had to be completely rewritten for me to even build a workspace.